combat/CombatSystem.cpp

```diff
@@ -523,24 +720,79 @@ namespace {
                 continue;
             ShipPartClass part_class = part->Class();
 
-            // get the attack power for each weapon part
-            float part_attack = 0.0f;
+            // direct weapon and fighter-related parts all handled differently...
+            if (part_class == PC_DIRECT_WEAPON) {
+                float part_attack = ship->CurrentPartMeterValue(METER_CAPACITY, part_name);
+                if (part_attack > 0.0f)
+                    retval.push_back(PartAttackInfo(part_class, part_name, part_attack));
+
+            } else if (part_class == PC_FIGHTER_HANGAR) {
+                // hangar max-capacity-modification effects stack, so only add capacity for each hangar type once
+                if (seen_hangar_part_types.find(part_name) == seen_hangar_part_types.end()) {
+                    available_fighters += ship->CurrentPartMeterValue(METER_CAPACITY, part_name);
+                    seen_hangar_part_types.insert(part_name);
+                }
+
+            } else if (part_class == PC_FIGHTER_BAY) {
+                part_fighter_launch_capacities[part_name] += ship->CurrentPartMeterValue(METER_CAPACITY, part_name);
+
+            } else if (part_class == PC_FIGHTER_WEAPON) {
+                fighter_attack = std::max(fighter_attack, ship->CurrentPartMeterValue(METER_CAPACITY, part_name));
+            }
+        }
+
+        if (available_fighters > 0 && !part_fighter_launch_capacities.empty()) {
+            for (std::multimap<std::string, int>::iterator launch_it = part_fighter_launch_capacities.begin();
```

Comment:
Gives a compiler error (MSVC 2010, Win7). Fixed by changing to ```std::map::iterator```.

If multimap is indeed intended, need to switch the type of ```part_fighter_launch_capacities```...